File: 2529. Maximum Count of Positive Integer and Negative Integer/main.go
package main

func maximumCount(nums []int) int {
	neg := 0
	for i := 0; i < len(nums); i++ {
		if nums[i] < 0 {
			neg++
		}
		if nums[i] > 0 {
			return max(neg, len(nums)-i)
		}
	}
	return neg
}

File: 2529. Maximum Count of Positive Integer and Negative Integer/main.py
from typing import List


class Solution:
    def maximumCount(self, nums: List[int]) -> int:
        neg = 0
        for i in range(len(nums)):
            if nums[i] < 0: neg += 1
            if nums[i] > 0:
                return max(neg, len(nums) - i)
        return neg